stable_baselines3/common/vec_env/vec_transpose.py

```diff
@@ -101,13 +103,16 @@ def step_wait(self) -> VecEnvStepReturn:
             if "terminal_observation" in infos[idx]:
                 infos[idx]["terminal_observation"] = self.transpose_observations(infos[idx]["terminal_observation"])
 
+        assert isinstance(observations, (np.ndarray, dict))
         return self.transpose_observations(observations), rewards, dones, infos
 
     def reset(self) -> Union[np.ndarray, Dict]:
```

Comment:
this is `VecEnv` api, and for various reasons (I repeated them across different issues/PR, I can link to them if you are interested), the main one being consistency, we will keep it as-is.

Reply:
Oh I would be indeed interested in the reasons. Might be worth putting them in a design decision record?

Reply:
It's a bit written in https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html#vecenv-api-vs-gym-api

The arguments are scattered around many issues but in short:
- for consistency between SB3 version (this is the main argument, already in the doc)
- for type checking (gym wrapper allow **kwargs)
- you don't need extra argument to achieve the same behavior (see https://github.com/DLR-RM/stable-baselines3/issues/650 and https://github.com/openai/gym/pull/2515#issuecomment-990712927)
- VecEnv are anyway different to Gym API (because they reset automatically among other things) so we should not pretend it is the same


Related issues: https://github.com/DLR-RM/stable-baselines3/pull/780#issuecomment-1277696805 and https://github.com/carlosluis/stable-baselines3/pull/1

Reply:
Thanks now I get it!